### DISCUSSION TITLE
Execute JavaScript - on button press - Close sidepanel

### DISCUSSION THREAD

**Discussion by ics460** [Clooos/Bubble-Card#572] [Questions about config, custom styles and templates] [answered]
My goal is to create a button that opens the sidebar, which can be done using button-card like this:

```
type: custom:button-card
icon: mdi:menu
tap_action:
  action: call-service
  service: |-
    [[[
      this.dispatchEvent(new Event('hass-toggle-menu', { bubbles: true, composed: true}));
      return none;
    ]]]
```

While this card supports JavaScript templating for styles, I haven't seen anything similar to the above. Is this currently possible?

**Comment by adamburgoyne** [comment]
Did you get anywhere with this? Seems like a pretty important feature given most people using this will have the header hidden. I tried to do the same as your post after checking out what mushroom-cards does for the menu chip but I've had no luck.

**Reply by ics460** [reply]
Nope, at the time I decided to keep using the mushroom-chips-card for a menu button but have since ditched it entirely as I now use this [material-rounded-theme](https://github.com/Nerwyn/material-rounded-theme) and no longer hide the header (since I like how it looks).

**Reply by adamburgoyne** [reply]
ah that's a shame but thanks. I've just hit the mushroom chip under my settings popup for the occasion I need the side bar on mobile. Hopefully it's something the developer can add as an action at some point.

**Comment by MrBearPresident** [comment, marked as answer]
You can do it like this: 
<details><summary>Proof gif</summary>

![Screen Recording 2024-12-30 210555](https://github.com/user-attachments/assets/d348c583-c53d-4988-9db8-a987e7dcae44)

</details>
<details><summary>Code</summary>

```yaml
type: custom:bubble-card
card_type: button
button_action:
  tap_action:
    action: none
button_type: name
name: Open Side panel
styles: |+
  .bubble-button-card{
    cursor: pointer !important;
  }

  ${(() => {
   if(!card.querySelector('.bubble-button-card').classList.contains('ExtraListener')){
     card.querySelector('.bubble-button-card').addEventListener("click", function() {
          this.dispatchEvent(new Event('hass-toggle-menu', { bubbles: true, composed: true}));
        });
      card.querySelector('.bubble-button-card').classList.add('ExtraListener') 
    }
  })()}
```

</details>
